File: apps/codemap-search/tests/fixtures/extract/sample.js
/** Fixture exercising JavaScript branch-sensitive extraction (TypeScript grammar today). */

/** A cache helper. */
export class Cache {
  /** Read a value. */
  read() {
    const key = "cache-key";
    return key;
  }

  /**
   * @deprecated use read instead
   */
  legacyRead() {
    return "legacy";
  }
}

/** An exported free function. */
export function publicLookup() {
  return 1;
}

function privateLookup() {
  return 0;
}

describe("Cache", () => {
  it("reads", () => {
    new Cache().read();
  });
});

File: apps/codemap-search/tests/fixtures/extract/sample.c
/** Fixture exercising C branch-sensitive extraction. */

#define MAX_CONNECTIONS 16

/** A connection record. */
struct Connection {
    int fd;
};

/** Open a connection and return a status message. */
const char *connection_open(void) {
    const char *status = "connection opened";
    return status;
}

/**
 * @deprecated use connection_open instead
 */
const char *connection_open_legacy(void) {
    return "legacy";
}

/* A file-local helper (static is not exported). */
static int internal_count(void) {
    return 0;
}
